src/components/map/MapboxMap.tsx

```diff
@@ -17,16 +17,26 @@ mapboxgl.workerClass = require("worker-loader!mapbox-gl/dist/mapbox-gl-csp-worke
 
 mapboxgl.accessToken = process.env.REACT_APP_MAPBOX_API_KEY as string;
 
-function mapOnLoad(map: mapboxgl.Map) {
+function mapOnLoad(map: mapboxgl.Map, dispatch: Dispatch<any>) {
   getEsriVectorSourceStyle(MapUrlResource.WHO_COUNTRY_VECTORTILES).then((style: mapboxgl.Style) => {
     addEsriLayersFromVectorSourceStyle(style, map);
-    const styleJson: any = map.getStyle();
+    const styleJson: Style = map.getStyle();
+    let CountryPolygonsMoved = false;
     if (styleJson && styleJson.layers) {
-      for (let layer of styleJson.layers) {
-        const t = layer["source-layer"];
-        if (t === "DISPUTED_AREAS") {
-          map.moveLayer("Countries", layer.id); // HACK for now, moves countries layer behind border once loaded.
-          break;
+      for (let layer of styleJson.layers as any) {
+        const source = layer["source-layer"];
+        if (source === "DISPUTED_AREAS") {
+          if (!CountryPolygonsMoved)
+          {
+            map.moveLayer("Countries", layer.id); // HACK for now, moves countries layer behind border once loaded.
+          }
+
+          map.on("mouseenter", layer.id, function (e: any) {
```

Comment:
I assume this logic here is to prevent the onhover country popup from showing up in disputed regions right?

Reply:
Yep!